whisper/src/aes_gcm/mod.rs

```diff
@@ -14,17 +14,54 @@
 // You should have received a copy of the GNU General Public License
 // along with Parity.  If not, see <http://www.gnu.org/licenses/>.
 
+//! # AES-GCM Module
+//!
+//! ### Overview
+//!
+//! Whisper's network protocol is secured using the Galois/Counter Mode of
+//! Operation (GCM) block cipher that provides confidentiality and message
+//! authentication assurance to inputs on each invocation (cf. [[2]]). 
+//!
+//! It provides authenticated encryption and signing ("sealing") of data,
+//! and authenticated decryption ("opening") of data that has been protected
+//! with Authenticated Encryption with Associated Data (AEAD).
+//! 
+//! It is constructed from the approved symmetric key block cipher Advanced
+//! Encryption Standard (AES) with a block size of 256-bits.
+//!
+//! [1]: https://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-38d.pdf
+//! [2]: http://luca-giuzzi.unibs.it/corsi/Support/papers-cryptography/gcm-spec.pdf
+
 use ring;
 
-/// AES GCM encryptor.
+/// Cryptographic secret key to seal the plaintext message, whose length is
+/// appropriate for the underlying block cipher.
+type SecretKey = [u8; 32];
+
+/// Initialization Vector (IV) is a nonce used during the sealing process.
+/// It must be distinct for each invocation of the encryption
+/// operation for a fixed value of the Secret Key. It is authenticated but it is
```

Comment:
Not sure what you mean by the nonce being authenticated tbh.

Reply:
It's actually written in http://luca-giuzzi.unibs.it/corsi/Support/papers-cryptography/gcm-spec.pdf on Page 5 of 43 as "The IV is authenticated, and it is not necessary
to include it in the AAD field."

Reply:
I guess I meant to say "I do not understand what is meant by that", not that it is wrong! If you understand it and feel that most code readers will understand is also, then feel free to leave it. If you, like me, suspect most readers will not, then remove it. Your call.